cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1711,6 +1711,8 @@ if(ARROW_WITH_PROTOBUF)
     set(ARROW_PROTOBUF_REQUIRED_VERSION "2.6.1")
   endif()
   resolve_dependency(Protobuf
+                     FORCE_ANY_NEWER_VERSION
```

Comment:
Can you add a comment explaning why we add `FORCE_ANY_NEWER_VERSION`?

Reply:
OK. Done.